pennylane/qnode_new/base.py

```diff
@@ -500,12 +500,12 @@ def evaluate(self, args, kwargs):
         """
         kwargs = self._default_args(kwargs)
 
-        if self.circuit is None or self.mutable:
-            self._construct(args, kwargs)
-
         # temporarily store the parameter values in the Variable class
         self._set_variables(args, kwargs)
 
+        if self.circuit is None or self.mutable:
+            self._construct(args, kwargs)
```

Comment:
Since variables were not being set until after construction, they weren't available during construction.